File: robocode2-server/src/main/java/net/robocode2/events/HitByBulletEvent.java
package net.robocode2.events;

import lombok.Value;
import net.robocode2.model.Bullet;

/**
 * Event sent when a bot is hit by a bullet.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class HitByBulletEvent implements Event {

	/** Turn number when event occurred */
	int turnNumber;

	/** Bullet that hit the bot */
	Bullet bullet;

	/** Damage dealt to the bot */
	double damage;

	/** New energy level of the bot after damage */
	double energy;
}

File: bot-api/python/tankroyale/botapi/events/HitByBulletEvent.py
from tankroyale.botapi import BulletState
import BotEvent


class HitByBulletEvent(BotEvent):
    bullet: BulletState
    damage: float
    energy: float

    def __init__(self, turn_number, bullet, damage, energy):
        super(turn_number)
        self.bullet = bullet
        self.damage = damage
        self.energy = energy

    def bullet(self):
        return self.bullet

    def damage(self):
        return self.damage

    def energy(self):
        return self.energy